src/models/subscriptions.js

```diff
@@ -18,18 +49,15 @@ function * subscribePairs (pairs, ledgersService, config) {
   yield ledgers.map((l) => subscribeLedger(l, ledgersService, config))
 }
 
-function * subscribeLedger (ledger, ledgersService, config) {
-  log.info('subscribing to ' + ledger)
-  yield ledgersService.subscribe(ledger, (resource, relatedResources) => {
-    co(function * () {
-      yield payments.updateTransfer(resource, relatedResources, ledgersService, config)
-    }).catch((err) => {
-      log.warn('error processing notification: ' + err)
-    })
-  })
+function * subscribeLedger (ledgerUri, ledgersService, config) {
+  log.info('subscribing to ' + ledgerUri)
+  const ledger = ledgersService.getLedger(ledgerUri)
+
+  ledger.connect()
```

Comment:
ledger.connect() returns a Promise, I think it should be returned (otherwise how do you catch an exception in `connect`). Also, does this need to be a generator function?